src/main.zig

```diff
@@ -4954,16 +4954,25 @@ fn cmdBuild(gpa: Allocator, arena: Allocator, args: []const []const u8) !void {
             if (debug_target) |triple| {
                 const target_query = try std.Target.Query.parse(.{
                     .arch_os_abi = triple,
+                    .cpu_features = "baseline", // See the comment below.
                 });
                 break :t .{
                     .result = std.zig.resolveTargetQueryOrFatal(target_query),
-                    .is_native_os = false,
-                    .is_native_abi = false,
+                    .is_native_os = target_query.isNativeOs(),
+                    .is_native_abi = target_query.isNativeAbi(),
                 };
             }
         }
+        // While it's tempting to use host machine info for performance reasons, this can
```

Comment:
I think the "performance reasons" are very dubious, the build runner is built in debug mode after all unless I'm mistaken and should definitely not be the bottleneck for compilation speed.

Reply:
I wouldn't expect it to be significant, but it isn't nothing either; even our self-hosted backends inspect target CPU features to emit more efficient code.